### PR TITLE
Added load/save support

### PR DESCRIPTION
**Pull request by yduchesne**
Hi, I've added support for load/save, with the accompanying tests. A few notes:

DockerClient.load allows loading the tarball of an image as a stream. An image name must be passed in, and the load implementation internally tags the just loaded image with the provided name. The load implementation is passed on Docker API's pull, which takes a fromSrc parameter, indicating it to treat the request's body as the .tar payload of the image.

DockerClient.save allows pulling the .tar stream of an image (and saving it locally or whatever).

### CODE REVIEW

**Comment by mattnworb** [comment]
Thanks for the contribution @yduchesne. This looks good to me :+1: two small requests though:

1. can you rebase and squash your commits to avoid having several each for "Added load/save support" and "Fixed CircleCI"? 
2. It's best to avoid re-ordering the import statements in a class when the change being made isn't related to the ordering of them, as this just reflects someone's editor's settings and is likely to flip-flop between various orderings based on different contributors.

File: src/main/java/com/spotify/docker/client/DefaultDockerClient.java [hunk 1]
```diff
@@ -88,28 +64,93 @@
...
 
 public class DefaultDockerClient implements DockerClient, Closeable {
 
+  /**
+   * Hack: this {@link ProgressHandler} is meant to capture the image ID of an
+   * image being loaded. Weirdly enough, Docker returns the ID of a newly created image
+   * in the status of a progress message.
+   * <p>
+   * The image ID is required to tag the just loaded image since, also weirdly enough,
+   * the pull operation with the <code>fromSrc</code> parameter does not suppor the <code>tag</code>
```
Comment (davidxia) [inline src/main/java/com/spotify/docker/client/DefaultDockerClient.java]: s/suppor/support/

File: src/main/java/com/spotify/docker/client/DefaultDockerClient.java [hunk 2]
```diff
@@ -641,7 +682,71 @@ public ContainerCreation commitContainer(final String containerId,
...
+  
+  @Override
+  public void load(String image, InputStream imagePayload)
+      throws DockerException, InterruptedException {
+    load(image, imagePayload, new LoggingPullHandler("image stream"));
+  }
+  
+  @Override
+  public void load(String image, InputStream imagePayload,
+      AuthConfig authConfig, ProgressHandler handler) throws DockerException,
```
Comment (davidxia) [inline src/main/java/com/spotify/docker/client/DefaultDockerClient.java]: small nit: we usually like to align these lines with the opening parentheses.

Comment (yduchesne) [inline src/main/java/com/spotify/docker/client/DefaultDockerClient.java]: Hi David, can you be more precise: which lines? Thanks.

Comment (davidxia) [inline src/main/java/com/spotify/docker/client/DefaultDockerClient.java]: line 694 can be aligned with the `S` in `load(String...)`. And thanks for the contribution :)

Comment (davidxia) [inline src/main/java/com/spotify/docker/client/DefaultDockerClient.java]: Same for various other lines.

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java [hunk 3]
```diff
@@ -298,8 +355,13 @@ public void testVersion() throws Exception {
 
   @Test
   public void testAuth() throws Exception {
-    final int statusCode = sut.auth(authConfig);
-    assertThat(statusCode, equalTo(200));
+    if (CIRCLECI) {
+      final int statusCode = sut.auth(authConfig);
+      assertThat(statusCode, anyOf(equalTo(200), equalTo(401)));
```
Comment (davidxia) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: Does circleCI really not support authenticating to docker hub anymore? If so, perhaps add a comment here and just ignore this test if running on circleCI with an `assumeFalse`?

File: src/main/java/com/spotify/docker/client/DefaultDockerClient.java [hunk 4]
```diff
@@ -823,7 +930,7 @@ public String build(final Path directory, final String name, final String docker
       return imageId;
     }
   }
-
+
```
Comment (davidxia) [inline src/main/java/com/spotify/docker/client/DefaultDockerClient.java]: I noticed a tiny white space here :)

**Comment by davidxia** [comment]
Other than my comments and @mattnworb's, :+1:

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java (hunk 3 above)
Comment (yduchesne) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: Hmm, that change is not from me. Maybe a merge glitch.

**Comment by yduchesne** [comment]
OK, closing this one and opening a new pull request (trying to update current request with changes but squashing does not seem to be taken into account)

**Pull request closed without merging**